### PR TITLE
Documentation/proxy: update due to Go 1.16 behavior change

### LINKED ISSUE #5624: HTTP_PROXY is not working
The `HTTP_PROXY` environmental variable listed in the documentation [here](https://github.com/grpc/grpc-go/blob/master/Documentation/proxy.md) is not working.

This can be easily verified with the hello_world example. Just run the following command:
```
$ HTTP_PROXY=10.0.0.1:8000 greeter_client -addr 10.0.0.2
```

It does not connect to the proxy at `10.0.0.1` but instead connects to the server `10.0.0.2` directly. However, the `HTTPS_PROXY` works as expected.

This seems due to the grpc code always sets the original URL scheme to `https`: https://github.com/grpc/grpc-go/blob/master/internal/transport/proxy.go#L43

And the underlying go library will only use the `HTTPS_PROXY` when the scheme is `https`: https://cs.opensource.google/go/x/net/+/master:http/httpproxy/proxy.go;l=129-130

Together, this means that the `HTTP_PROXY` can never be used.

Either the grpc-code should be fixed to support the `HTTP_PROXY` correctly or the [documentation](https://github.com/grpc/grpc-go/blob/master/Documentation/proxy.md) should be updated to mention that `HTTP_PROXY` is not supported. 

### What version of gRPC are you using? 
Tested in master but probably exist in previous versions as well.

### What version of Go are you using (`go version`)? 
1.19

### What operating system (Linux, Windows, …) and version? 
MacOS

### What did you do?
Can easily verify with the example `HTTP_PROXY=10.0.0.1:8000 greeter_client -addr 10.0.0.2`

### What did you expect to see?
The greeter_client connects to the HTTP_PROXY at `10.0.0.1`

### What did you see instead?
The greeter_client bypassed the HTTP_PROXY and connects directly to the address `10.0.0.2`

### PR DESCRIPTION
**Pull request by dfawley**
Fixes #5624 

RELEASE NOTES: n/a

### CODE REVIEW

File: Documentation/proxy.md [hunk 1]
```diff
@@ -12,4 +13,4 @@ connection before giving it to gRPC.
 
 If the default proxy doesn't work for you, replace the default dialer with your
 custom proxy dialer. This can be done using
-[`WithDialer`](https://godoc.org/google.golang.org/grpc#WithDialer).
\ No newline at end of file
+[`WithDialer`](https://godoc.org/google.golang.org/grpc#WithDialer).
```
Comment (zasweq) [inline Documentation/proxy.md:16]: What is the diff here lol?

File: Documentation/proxy.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
 # Proxy
 
 HTTP CONNECT proxies are supported by default in gRPC. The proxy address can be
-specified by the environment variables HTTP_PROXY, HTTPS_PROXY and NO_PROXY (or
-the lowercase versions thereof).
+specified by the environment variables `HTTPS_PROXY` and `NO_PROXY`.  Before Go
+1.16, if the `HTTPS_PROXY` environment variable is unset, `HTTP_PROXY` will be
+used instead.  (Note that these environment variables are case insensitive.)
```
Comment (zasweq) [inline Documentation/proxy.md:6]: unless it's a CGI environment ;P. Jk, I think that's too specific to mention.

**Review by zasweq — APPROVED** [review]
LGTM.

File: Documentation/proxy.md (hunk 1 above)
Comment (dfawley) [reply Documentation/proxy.md:16]: Is there a whitespace at the end of the line that the diff viewer is hiding or something? I'm not sure; I didn't touch it but my editor must have changed something? It goes away when hiding whitespace diffs.

**Pull request merged**